src/workers.ts

```diff
@@ -1,18 +1,44 @@
-'use strict'
-const { isMainThread, parentPort } = require('worker_threads')
-const { AsyncResource } = require('async_hooks')
+import { isMainThread, parentPort } from 'worker_threads'
+
+import { AsyncResource } from 'async_hooks'
+
+export interface ThreadWorkerOptions {
+  /**
+   * Max time to wait tasks to work on (in ms), after this period the new worker threads will die.
+   *
+   * @default 60.000 ms
+   */
+  maxInactiveTime?: number
+  /**
+   * `true` if your function contains async pieces, else `false`.
+   *
+   * @default false
+   */
+  async?: boolean
+}
 
 /**
- * An example worker that will be always alive, you just need to extend this class if you want a static pool.<br>
- * When this worker is inactive for more than 1 minute, it will send this info to the main thread,<br>
- * if you are using DynamicThreadPool, the workers created after will be killed, the min num of thread will be guaranteed
- * @author Alessandro Pio Ardizio
+ * An example worker that will be always alive, you just need to **extend** this class if you want a static pool.
+ *
+ * When this worker is inactive for more than 1 minute, it will send this info to the main thread,
+ * if you are using DynamicThreadPool, the workers created after will be killed, the min num of thread will be guaranteed.
+ *
+ * @author [Alessandro Pio Ardizio](https://github.com/pioardi)
  * @since 0.0.1
  */
-class ThreadWorker extends AsyncResource {
-  constructor (fn, opts) {
+export class ThreadWorker<Data = any, Response = any> extends AsyncResource {
+  protected readonly maxInactiveTime: number
+  protected readonly async: boolean
+  protected lastTask: number
+  protected readonly interval: NodeJS.Timeout
+  protected parent: any
+
+  public constructor (
+    fn: (data: Data) => Response,
+    public readonly opts: ThreadWorkerOptions = {}
+  ) {
     super('worker-thread-pool:pioardi')
-    this.opts = opts || {}
```

Comment:
Why did we removed this line ? Any particular reason ?

Reply:
It was just moved to line 38  
<img src="https://user-images.githubusercontent.com/7195563/107203720-10d4f900-69fc-11eb-9e01-fdabefcbf057.png" width="400px"/>

Reply:
Thanks @Shinigami92